### DISCUSSION TITLE
Vue: katex.renderToString increases the height of the page

### DISCUSSION THREAD

**Discussion by kobayashi-dot** [KaTeX/KaTeX#3098] [Q&A] [answered]
I have a vue component like this:

```
<template>
    <div id="wrapper">
        <div id="container">
            <card v-for="n of Array(100).keys()" :key="n"></card>
        </div>
    </div>
</template>
<style scoped>
#container {
    max-width: 50%;
    overflow: scroll;
    height: 50vh;
    background-color: orange;
}

#wrapper {
    height: 100vh;
}
</style>
```

This generates 100 cards in the container. The Card is:

```
<template>
        <div ref="content">$$e^{i\pi}+1=0$$</div>
</template>
<script>
const katex = require('katex')
export default {
    mounted(){
        this.$refs.content.innerHTML = this.$refs.content.innerHTML.replace(/\$\$([^]*?)\$\$/g, function(outer, inner) {
            return katex.renderToString(inner, { displayMode: true})
        })
    },
}
</script>
```

However, this will create a blank space at the bottom of the page. How to remove it?

https://user-images.githubusercontent.com/64967842/123539363-5a886b00-d774-11eb-8322-5da89761ae85.mov

**Comment by edemaine** [comment, marked as answer]
Hmm, I don't see why KaTeX would cause that.  Does element inspector reveal anything about the generated HTML/CSS?  I suspect there is a different cause...

**Reply by kobayashi-dot** [reply]
If I change the line `return katex.renderToString(inner, { displayMode: true})` to `return 'Hello'`, the problem does not occur.

**Reply by edemaine** [reply]
What if you replace it with `return '<div style="padding: 1em">Hello</div>'`?

**Reply by kobayashi-dot** [reply]
In that case, the problem will not occur.

**Reply by edemaine** [reply]
Can you try `return katex.renderToString(inner, { displayMode: true, output: 'html'})`?  My theory is that the MathML, which renders invisibly by constraining it to a 1x1 pixel box at some absolute coordinates, is causing the problem.

**Reply by kobayashi-dot** [reply]
It's fixed! Thank you.
